### PR TITLE
Support N dim Tensor input in PrepRankingData job

### PR DESCRIPTION
**Pull request by plliao**
The PR supports N-dimensional dense/sparse tensor input in the PrepRankingData job.

Why?
We would like to extend the use case of 2 dimension tensor into N dimension tensor for both dense and sparse tensor. The change is backward compatible with SparseVector.

Design decision for the N-dimensional schema:
The schema of N-dimensional tensor extends what existed in the Avro2TF.
For N-dimensional dense tensor, the schema is a nested array.
For example, ArrayType[ArrayType[ArrayType[Double]]] for 3D double tensor.

For N-dimensional sparse tensor, the schema follows "generateSchemaForMultiDimSparseFeature" function in the PrepRankingData job.
StructType:
  StructField: indices0
  StructField: indices1
  ...
  StructField: indicesN
  StructField: values

### CODE REVIEW

File: avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/PrepRankingData.scala [hunk 1]
```diff
@@ -122,12 +123,16 @@ object PrepRankingData {
...
+    val documentFeaturesWithSparseTensor = documentFeatures.filter(
       it =>
-        CommonUtils.isArrayOfSparseVector(truncateDf.schema(it).dataType))
+        CommonUtils.isArrayOfSparseTensor(truncateDf.schema(it).dataType))
 
-    documentFeaturesWithSpVec.foreach { it =>
-      truncateDf = truncateDf.withColumn(it, twoDimensionSparseFeature(col(it)))
+    val sparseTensorFeatureNameAndRanks = documentFeaturesWithSparseTensor.map(
+      it => (it, selectDf.schema(it).dataType
+        .asInstanceOf[StructType].fieldNames.length - 1)) // decrease the dimension of values
```
Comment (zhangxuhong) [inline avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/PrepRankingData.scala]: can we merge them?

**Review by zhangxuhong — APPROVED** [review]
overall LGTM.

**Review by cyzhangchenya — COMMENTED** [review]
High level LGTM.
One suggestion, the documentation and tutorial need to be updated accordingly per this change (also previous changes...).
If you have time, please help to do so. (Please also check with Yiming.)

File: avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/PrepRankingData.scala (hunk 1 above)
Comment (plliao) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/PrepRankingData.scala]: sure! done.

**Pull request merged**